cmake/BLTMacros.cmake

```diff
@@ -290,7 +287,7 @@ endmacro(blt_register_library)
 macro(blt_add_library)
 
     set(singleValueArgs NAME OUTPUT_NAME OUTPUT_DIR HEADERS_OUTPUT_SUBDIR SHARED)
-    set(multiValueArgs SOURCES HEADERS DEPENDS_ON)
+    set(multiValueArgs SOURCES HEADERS INCLUDES DEFINES DEPENDS_ON)
```

Comment:
Does it make sense to pull out some of this logic into a private macro, e.g. if it duplicates logic that is in ``blt_register_library``?

Reply:
My idea was to get blt_add_library fully flushed out then make blt_register_library only for external libraries when I can get the imported targets working.  The mixing of the two idioms is will not work in the future I would expect.  Not to mention the confusion I think it creates.  Maybe the name needs to change.  Like blt_import_target